internal/backend/runtime/omni/controllers/omni/internal/exposedservice/reconciler.go

```diff
@@ -136,32 +120,13 @@ func (reconciler *Reconciler) reconcileService(ctx context.Context, exposedServi
 		explicitAliasOpt = optional.Some(explicitAlias)
 	}
 
-	updatedService, err := safe.WriterModifyWithResult(ctx, reconciler.r, exposedService, func(res *omni.ExposedService) error {
-		return reconciler.updateExposedService(res, explicitAliasOpt, port, label, icon, logger)
-	})
-	if err != nil {
+	if err = reconciler.updateExposedService(exposedService, explicitAliasOpt, port, label, icon, logger); err != nil {
 		return false, fmt.Errorf("error updating exposed service: %w", err)
 	}
 
-	updatedVersion := updatedService.Metadata().Version()
-	cluster, _ := updatedService.Metadata().Labels().Get(omni.LabelCluster)
-	hasExplicitAlias := updatedService.TypedSpec().Value.HasExplicitAlias
-
-	if !updatedVersion.Equal(version) {
-		logger.Info("updated exposed service",
-			zap.Uint64("version", updatedVersion.Value()),
-			zap.String("cluster", cluster),
-			zap.Bool("has_explicit_alias", hasExplicitAlias),
-			zap.String("url", updatedService.TypedSpec().Value.Url),
-			zap.Uint32("port", updatedService.TypedSpec().Value.Port),
-			zap.String("label", updatedService.TypedSpec().Value.Label),
-			zap.String("error", updatedService.TypedSpec().Value.Error),
-		)
-	}
-
-	alias, _ := updatedService.Metadata().Labels().Get(omni.LabelExposedServiceAlias)
-	reconciler.usedAliases[alias] = updatedService.Metadata().ID()
-	reconciler.exposedServices[updatedService.Metadata().ID()] = updatedService
```

Comment:
Moved into the controller, as we do not do actual operations on the state in this component anymore. It is now side-effect free.